### PR TITLE
Do not count in pipelined task when calculating ready tasks number

### PR DESCRIPTION
**Pull request by hex108**
**What this PR does / why we need it**:
Do not count in pipelined task when calculating ready tasks number because we could not make a promise that they will get allocated on next schedule. They might not be allocated for following cases:
* Pipelined resources are allocated to other pods by other schedulers.
* A job with higher priority comes in, and takes pipelined resources.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
Do not count in pipelined task when calculating ready tasks number.
```

### CODE REVIEW

**Comment by hex108** [comment]
However we should count in pipelined tasks when preempting. Fix the problem.

**Comment by hex108** [comment]
There seems no errors in the travis-ci log.

/retest

**Comment by hex108** [comment]
@k82cn How could we trigger retest like bots in kubernetes repo?

**Comment by k82cn** [comment]
sure :)

**Comment by fejta** [comment]
What job are you wanting to retest? I don't see a failed prowjob in the status contexts. Do you want to convert the travis workflow to a prow job?

**Comment by hex108** [comment]
@fejta Thanks for it! We need to convert it to a prow job.

File: pkg/scheduler/actions/allocate/allocate.go [hunk 1]
```diff
@@ -184,7 +184,7 @@ func (alloc *allocateAction) Execute(ssn *framework.Session) {
 				break
 			}
 
-			if ssn.JobReady(job) {
+			if ssn.JobReadyForRun(job) {
```
Comment (k82cn) [inline pkg/scheduler/actions/allocate/allocate.go]: prefer to `JobReady` and `JobPipelined or JobWaiting`

File: pkg/scheduler/api/job_info.go [hunk 2]
```diff
@@ -349,3 +349,57 @@ func (ji *JobInfo) FitError() string {
...
+			status == Succeeded {
+			occupid = occupid + len(tasks)
+		}
+	}
+
+	return int32(occupid)
+}
+
+// readyTaskNumForRun return the number of tasks that are ready for preempt.
+func (ji *JobInfo) ReadyTaskNumForPreempt() int32 {
```
Comment (k82cn) [inline pkg/scheduler/api/job_info.go]: prefer to `WaitingTaskNum` which only includes `Pipelined`; the `Releasing` resource maybe triggerred by reclaim.

**Comment by k82cn** [comment]
LGTM overall, just some suggestion on the func name.

**Comment by hex108** [comment]
@k82cn PTAL

**Pull request merged**